azext_edge/edge/providers/rpsaas/adr/asset_endpoint_profiles.py

```diff
@@ -78,9 +80,19 @@ def create(
         # Properties
         properties = {"endpointProfileType": endpoint_profile_type}
 
+        configuration = None
         if endpoint_profile_type == AEPTypes.opcua.value:
-            properties["additionalConfiguration"] = _build_opcua_config(**additional_configuration)
-        # TODO: add other connector types in
+            configuration = _build_opcua_config(**kwargs)
+        elif kwargs.get("additional_configuration"):  # custom type
+            configuration = kwargs["additional_configuration"]
+            try:
+                logger.debug("Processing additional configuration.")
+                configuration = read_file_content(configuration)
+            except FileOperationError:
+                logger.debug("Given additional configuration is not a file.")
+                pass
+        properties["additionalConfiguration"] = configuration
+
         _update_properties(
```

Comment:
`_update_properties()` has an `additional_configuration` param that assigns `properties["additionalConfiguration"]` , but that also happening higher up. Can that be consolidated?